### PR TITLE
wazevo: passes simd_boolean spec tests

### PR DESCRIPTION
**Pull request by evacchi**
continues #1496

passes `simd_boolean` spec tests, adds related instructions: `addv`, `uminv`, `umaxp` `dup`, `ext`, `zip`, `sshr`, `cmeq(zero)`

### CODE REVIEW

File: internal/engine/wazevo/backend/isa/arm64/lower_instr.go [hunk 1]
```diff
@@ -395,6 +399,189 @@ func (m *machine) LowerInstr(instr *ssa.Instruction) {
...
+	case ssa.VecLaneI8x16:
+		sshr := m.allocateInstr()
+		sshr.asVecShiftImm(vecOpSshr, v1, rm, operandShiftImm(7), vecArrangement16B)
+		m.insert(sshr)
+
+		// m.lowerConstantI64(r0.nr(), 0x0008000400020001)
+		m.insertMOVZ(r0.nr(), 0x0201, 0, true)
+		m.insertMOVK(r0.nr(), 0x0804, 1, true)
+		m.insertMOVK(r0.nr(), 0x2010, 2, true)
+		m.insertMOVK(r0.nr(), 0x8040, 3, true)
```
Comment (evacchi) [inline internal/engine/wazevo/backend/isa/arm64/lower_instr.go:474]: this could be refactored into a lower/splat constant

File: internal/engine/wazevo/backend/isa/arm64/lower_instr.go [hunk 2]
```diff
@@ -395,6 +399,189 @@ func (m *machine) LowerInstr(instr *ssa.Instruction) {
...
+
+		dup := m.allocateInstr()
+		dup.asVecDup(v0, r0, vecArrangement2D)
```
Comment (mathetake) [inline internal/engine/wazevo/backend/isa/arm64/lower_instr.go:477]: what if v0 is used somewhere else? (seems a same bug as we discussed) Instead use the temporary value.

File: internal/engine/wazevo/backend/isa/arm64/lower_instr.go [hunk 3]
```diff
@@ -395,6 +399,189 @@ func (m *machine) LowerInstr(instr *ssa.Instruction) {
...
+		m.insert(dup)
+
+		and := m.allocateInstr()
+		and.asVecRRR(vecOpAnd, v1, v1, v0, vecArrangement16B)
```
Comment (mathetake) [inline internal/engine/wazevo/backend/isa/arm64/lower_instr.go:481]: same

File: internal/engine/wazevo/backend/isa/arm64/lower_instr.go (hunk 1 above)
Comment (evacchi) [reply internal/engine/wazevo/backend/isa/arm64/lower_instr.go]: I'll leave it as-is for now.

File: internal/engine/wazevo/backend/isa/arm64/lower_instr.go [hunk 4]
```diff
@@ -395,6 +399,272 @@ func (m *machine) LowerInstr(instr *ssa.Instruction) {
 	m.FlushPendingInstructions()
 }
 
+func (m *machine) lowerVcheckTrue(instr *ssa.Instruction) {
```
Comment (mathetake) [inline internal/engine/wazevo/backend/isa/arm64/lower_instr.go:402]: can we have a unit test on this

File: internal/engine/wazevo/backend/isa/arm64/lower_instr.go [hunk 5]
```diff
@@ -395,6 +399,272 @@ func (m *machine) LowerInstr(instr *ssa.Instruction) {
...
+	x, lane := instr.ArgWithLane()
+	rm := m.getOperand_NR(m.compiler.ValueDefinition(x), extModeNone)
+	rd := operandNR(m.compiler.VRegOf(instr.Return()))
+
+	r0 := operandNR(m.compiler.AllocateVReg(regalloc.RegTypeOf(ssa.TypeI64)))
+	v0 := operandNR(m.compiler.AllocateVReg(regalloc.RegTypeOf(ssa.TypeV128)))
+	v1 := operandNR(m.compiler.AllocateVReg(regalloc.RegTypeOf(ssa.TypeV128)))
+
+	switch lane {
+
```
Comment (mathetake) [inline internal/engine/wazevo/backend/isa/arm64/lower_instr.go:475]:
```suggestion
```

File: internal/engine/wazevo/ssa/instructions.go [hunk 6]
```diff
@@ -913,8 +913,8 @@ var instructionReturnTypes = [opcodeEnd]returnTypesFn{
 	OpcodeVbnot:      returnTypesFnV128,
 	OpcodeVbandnot:   returnTypesFnV128,
 	OpcodeVbitselect: returnTypesFnV128,
-	OpcodeVanyTrue:   returnTypesFnV128,
-	OpcodeVallTrue:   returnTypesFnV128,
+	OpcodeVanyTrue:   returnTypesFnSingle,
+	OpcodeVallTrue:   returnTypesFnSingle,
```
Comment (mathetake) [inline internal/engine/wazevo/ssa/instructions.go:917]: what's the rationale for these changes

Comment (mathetake) [reply internal/engine/wazevo/ssa/instructions.go:917]: why can't you use `returnTypesFnI32`

**Review by mathetake — APPROVED** [review]
talked offline - let's merge once `DeterministicCompilationVerifierEnabled=true` passes

File: internal/engine/wazevo/ssa/instructions.go (hunk 6 above)
Comment (evacchi) [reply internal/engine/wazevo/ssa/instructions.go:917]: I did not notice there was a proper `returnTypesFnI32`
